controller.py

```diff
@@ -0,0 +1,41 @@
+import functools
+
+class Controller(object):
+    __error = False
```

Comment:
What is the rationale behind defining __error as a class attribute rather than an instance attribute?